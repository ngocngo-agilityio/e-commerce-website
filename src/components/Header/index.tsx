// Libs
import { Box, Container, HStack, Show, Hide, Flex } from '@chakra-ui/react';

// Assets
import { HamburgerIcon } from '@assets';

// APIs
import { getCartItems } from '@apis';

// Auth configs
import { auth } from '@configs';

// Components
import { Logo, Navigation, ShoppingCart, UserProfile } from '@components';

const Header = async (): Promise<JSX.Element> => {
  const session = await auth();

  const userId = session?.user?.id || '';

  // Get my cart
  const { data: myCart } = await getCartItems(userId);

  const { cartItems = [] } = myCart || {};
  const cartItemQuantity = cartItems.length || 0;

  return (
    <Container
      as="header"
      display="flex"
      flexDir="column"
      gap={2}
      pt="34px"
      pb={{ base: 5, md: 4 }}
    >
      <Flex alignContent="center" justifyContent="space-between">
        <HStack spacing={{ md: '100px', lg: '150px' }} wrap="wrap">
          <Logo />
          <Show above="md">
            <Navigation />
          </Show>
        </HStack>

        <HStack spacing="25px" alignContent="center">
          <UserProfile />
          <ShoppingCart cartItemQuantity={cartItemQuantity} />

          <Box cursor="pointer">
            <HamburgerIcon />
          </Box>
        </HStack>
      </Flex>

      <Hide above="md">
        <Navigation />
      </Hide>
    </Container>
  );
};

export default Header;
