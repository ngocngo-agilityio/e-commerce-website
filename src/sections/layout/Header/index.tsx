// Libs
import {
  Box,
  Container,
  HStack,
  Show,
  Hide,
  Flex,
  Skeleton,
} from '@chakra-ui/react';
import { Suspense } from 'react';

// Assets
import { HamburgerIcon } from '@assets';

// Components
import { Logo, Navigation, UserProfile } from '@components';

// Sections
import { ShoppingCart } from '@sections';

const Header = (): JSX.Element => (
  <Container
    as="header"
    display="flex"
    flexDir="column"
    gap={2}
    pt="34px"
    pb={{ base: 5, md: 4 }}
  >
    <Flex alignContent="center" justifyContent="space-between">
      <HStack spacing={{ md: '100px', lg: '150px' }} wrap="wrap">
        <Logo />
        <Show above="md">
          <Navigation />
        </Show>
      </HStack>

      <HStack spacing="25px" alignContent="center">
        <UserProfile />
        <Suspense fallback={<Skeleton w={6} h={6} />}>
          <ShoppingCart />
        </Suspense>

        <Box cursor="pointer">
          <HamburgerIcon />
        </Box>
      </HStack>
    </Flex>

    <Hide above="md">
      <Navigation />
    </Hide>
  </Container>
);
export default Header;
