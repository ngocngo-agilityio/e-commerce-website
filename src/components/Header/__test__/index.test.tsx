// Libs
import { render } from '@testing-library/react';

// Apis
import { getCartItems } from '@apis';

// Mocks
import { MOCK_CART_ITEMS } from '@mocks';

// Components
import Header from '..';

jest.mock('@apis', () => ({
  ...jest.requireActual('@apis'),
  getCartItems: jest.fn(),
}));

describe('Header component', () => {
  beforeEach(() => {
    (
      getCartItems as jest.MockedFunction<typeof getCartItems>
    ).mockResolvedValue({ data: MOCK_CART_ITEMS });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render Header successfully', async () => {
    (
      getCartItems as jest.MockedFunction<typeof getCartItems>
    ).mockResolvedValue({ data: [] });

    const { container } = render(await Header());

    expect(container).toBeInTheDocument();
  });

  test('should match snapshot for Header', async () => {
    const { container } = render(await Header());

    expect(container).toMatchSnapshot();
  });
});
