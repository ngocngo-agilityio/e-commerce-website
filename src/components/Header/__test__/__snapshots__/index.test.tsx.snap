// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot for Header 1`] = `
<div>
  <header
    class="chakra-container css-0"
  >
    <div
      class="css-171xvke"
    >
      <div
        class="chakra-stack css-1ji2e0z"
      >
        <h1
          class="chakra-heading css-jouhmg"
        >
          <a
            href="/"
          >
            NorthStar
          </a>
        </h1>
        <nav
          class="css-7dibxd"
        >
          <a
            href="#"
          >
            <span
              class="chakra-text css-1fd5thk"
            >
              Home
            </span>
          </a>
          <a
            href="#"
          >
            <span
              class="chakra-text css-1fd5thk"
            >
              About
            </span>
          </a>
          <a
            href="#"
          >
            <span
              class="chakra-text css-1fd5thk"
            >
              Contact us
            </span>
          </a>
        </nav>
      </div>
      <div
        class="chakra-stack css-erq1zg"
      >
        <button
          aria-controls="menu-list-:r3:"
          aria-expanded="false"
          aria-haspopup="menu"
          aria-label="profile-menu"
          class="chakra-menu__menu-button css-1eqi7qs"
          id="menu-button-:r3:"
        >
          <span
            class="css-xl71ch"
          >
            <svg
              class="chakra-icon css-1t7rsys"
              focusable="false"
              viewBox="0 0 24 25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 21.4761V19.4761C20 18.4152 19.5786 17.3978 18.8284 16.6476C18.0783 15.8975 17.0609 15.4761 16 15.4761H8C6.93913 15.4761 5.92172 15.8975 5.17157 16.6476C4.42143 17.3978 4 18.4152 4 19.4761V21.4761"
                stroke="#1D1D1D"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M12 11.4761C14.2091 11.4761 16 9.68521 16 7.47607C16 5.26694 14.2091 3.47607 12 3.47607C9.79086 3.47607 8 5.26694 8 7.47607C8 9.68521 9.79086 11.4761 12 11.4761Z"
                stroke="#1D1D1D"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </button>
        <div
          class="css-79elbk"
        >
          <a
            aria-label="cart-icon"
            href="/cart"
          >
            <svg
              class="chakra-icon css-1t7rsys"
              focusable="false"
              viewBox="0 0 24 25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2.47607L3 6.47607V20.4761C3 21.0065 3.21071 21.5152 3.58579 21.8903C3.96086 22.2654 4.46957 22.4761 5 22.4761H19C19.5304 22.4761 20.0391 22.2654 20.4142 21.8903C20.7893 21.5152 21 21.0065 21 20.4761V6.47607L18 2.47607H6Z"
                stroke="#1D1D1D"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M3 6.47607H21"
                stroke="#1D1D1D"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M16 10.4761C16 11.5369 15.5786 12.5544 14.8284 13.3045C14.0783 14.0546 13.0609 14.4761 12 14.4761C10.9391 14.4761 9.92172 14.0546 9.17157 13.3045C8.42143 12.5544 8 11.5369 8 10.4761"
                stroke="#1D1D1D"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
        </div>
        <div
          class="css-4g6ai3"
        >
          <svg
            class="chakra-icon css-1t7rsys"
            focusable="false"
            viewBox="0 0 24 25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12.4761H21"
              stroke="#1D1D1D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M3 6.47607H21"
              stroke="#1D1D1D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M3 18.4761H21"
              stroke="#1D1D1D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </div>
  </header>
</div>
`;
